api/src/main/java/io/grpc/TlsChannelCredentials.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+import java.util.Collections;
+import java.util.Set;
+
+/**
+ * TLS credentials, providing server authentication and encryption. Consumers of this credential
+ * must verify they understand the configuration via the {@link #incomprehensible} method. Unless
+ * overridden by a {@code Feature}, server verification should use customary default root
+ * certificates.
+ */
+@ExperimentalApi("TODO")
+public final class TlsChannelCredentials extends ChannelCredentials {
+  /** Use TLS with its defaults. */
+  public static ChannelCredentials create() {
+    return newBuilder().build();
+  }
+
+  TlsChannelCredentials(Builder builder) {}
+
+  /**
+   * Returns an empty set if this credential can be adequately understood via
+   * the features listed, otherwise returns a hint of features that are lacking
+   * to understand the configuration to be used for manual debugging.
+   *
+   * <p>An "understood" feature does not imply the caller is able to fully
+   * handle the feature. It simply means the caller understands the feature
+   * enough to use the appropriate APIs to read the configuration. The caller
+   * may support just a subset of a feature, in which case the caller would
+   * need to look at the configuration to determine if only the supported
+   * subset is used.
+   *
+   * <p>This method may not be as simple as a set difference. There may be
+   * multiple features that can independently satisfy a piece of configuration.
+   * If the configuration is incomprehensible, all such features would be
+   * returned, even though only one may be necessary.
+   *
+   * <p>An empty set does not imply that the credentials are fully understood.
+   * There may be optional configuration that can be ignored if not understood.
+   *
+   * <p>Since {@code Feature} is an {@code enum}, {@code understoodFeatures}
+   * should generally be an {@link java.util.EnumSet}. {@code
+   * understoodFeatures} will not be modified.
+   */
+  public Set<Feature> incomprehensible(Set<Feature> understoodFeatures) {
```

Comment:
Couple of things:

- something I have been thinking: how about using a more commonly used term like `unsupported` instead of `incomprehensible` ? I understand there is some difference between unsupported and not-even-understood features in terms of their meaning but that's immaterial here. If a transport doesn't understand or support a certain feature that transport is going to be skipped.

- there are currently no unit tests to test this functionality e.g. `incomprehensible` returns a non-empty set so a transport is skipped.

Reply:
How about `unrecognized` as it is still better (IMO) than `incomprehensible`? Also call the param `requiredFeatures` and describe it using the `@param` tag in the method Javadoc. Just a suggestion.

Reply:
`unrecognized` makes it sound like _TlsChannelCredentials_ doesn't understand some feature, and `requiredFeatures` is that all the more.

Some potential other options for the method name: `comprehensible`, `understandable`, `missingFeatures`, `neededFeatures`, maybe something along the lines of "safe", or maybe `compatible`/`incompatible`

Reply:
Like I said earlier it is just a suggestion for what I think would be a better name than `incomprehensible` but I understand it's subjective so it's not a big deal. It will be good to have some description (something to the effect that the caller i.e. transport will pass the features that it "understands")  with the param tag for the `understoodFeatures` param.